lua/fidget/progress/handle.lua

```diff
@@ -0,0 +1,145 @@
+local notification = require("fidget.notification")
+
+local NEXT_ID = 0
+local PREFIX = "fidget-"
+
+local function next_token()
+  NEXT_ID = NEXT_ID + 1
+  return PREFIX .. NEXT_ID
```

Comment:
nit: `PREFIX .. tostring(NEXT_ID)` rather than `PREFIX .. NEXT_ID`, to be explicit about the string coercion